scipy/stats/_distn_infrastructure.py

```diff
@@ -3538,13 +3581,30 @@ def __init__(self, a=0, b=inf, name=None, badvalue=None,
         self.qvals = np.cumsum(self.pk, axis=0)
 
         self.shapes = ' '   # bypass inspection
+
         self._construct_argparser(meths_to_inspect=[self._pmf],
                                   locscale_in='loc=0',
                                   # scale=1 for discrete RVs
                                   locscale_out='loc, 1')
 
+        self._attach_argparser_methods()
+
         self._construct_docstrings(name, longname, extradoc)
 
+    def __getstate__(self):
+        dct = self.__dict__.copy()
+
+        # these methods will be remade in __setstate__
+        # __setstate__ is supplied by rv_discrete
+        attrs = ["_parse_args", "_parse_args_stats", "_parse_args_rvs"]
+        [dct.pop(attr, None) for attr in attrs]
+
+        return dct
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+        self._attach_argparser_methods()
```

Comment:
The comment above says it's supplied by rv_discrete, so no needed here (or the comment should be fixed)

Reply:
Changed in a3ea650cd6240aad3ed321766b7253ac17ecc6a8